core/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -182,6 +182,37 @@ class DeltaTable private[tables](
     executeDelete(None)
   }
 
+  /**
+   * Optimize data in the table that match the given `condition`.
+   *
+   * @param condition Boolean SQL expression
+   *
+   * @since 1.2.0
+   */
+  def optimize(condition: String): Unit = {
```

Comment:
SQL equivalent of Optimize returns DataFrame containing the operation metrics. We can do the same here (look at the vacuum for examples).

Reply:
Do you know offhand the best way to convert the `Seq[Row]` to a DataFrame? I mostly only use the Python API. There's a `createDataFrame(rows: List[Row], schema: [StructType])`, not sure if there's a way to use the `output` to create that schema or if I should just manually define that in the class as well

Reply:
And it looks like `executeVacuum` also just returns `sparkSession.emptyDataFrame`

Reply:
It can be done similar to `executeGenerate` [here](https://github.com/delta-io/delta/blob/master/core/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala#L57). Instead of using the `OptimizeExecutor`, use the `DeltaCommand` implementation `OptimizeTableCommand`. This also removes the changes where the partition filter has is moved from `OptimizeTableCommand` to `OptimizeExecutor`.

Reply:
I didn't try to go that route initially because it seemed weird to call a function on DeltaTable that then has to generate the table ID to pass to another function just to it can re-look up the DeltaLog again. And same for the expression, I wanted to add passing a Column expression and it seemed weird to have to convert the Column back to SQL just so it can be parsed into an expression again

Reply:
Got that route working by creating the table identifier like the generate command does